tests/unit/src/main/scala/tests/BaseAmmoniteSuite.scala

```diff
@@ -0,0 +1,287 @@
+      """main.sc:15:25: error: not found: type Fooz
+        |val decodedFoo = decode[Fooz](json)
+        |                        ^^^^
+        |""".stripMargin
+    for {
+      _ <- server.initialize(
+        s"""
+           |/metals.json
+           |{
+           |  "a": {
+           |    "scalaVersion": "$scalaVersion"
+           |  }
+           |}
+           |/main.sc
+           | // scala $scalaVersion
+           |import $$ivy.`io.circe::circe-core:0.12.3`
+           |import $$ivy.`io.circe::circe-generic:0.12.3`
+           |import $$ivy.`io.circe::circe-parser:0.12.3`
+           |import io.circe._, io.circe.generic.auto._, io.circe.parser._, io.circe.syntax._
+           |
+           |sealed trait Foo
+           |case class Bar(xs: Vector[String]) extends Foo
+           |case class Qux(i: Int, d: Option[Double]) extends Foo
+           |
+           |val foo: Foo = Qux(13, Some(14.0))
+           |
+           |val json = foo.asJson.noSpaces
+           |
+           |val decodedFoo = decode[Fooz](json)
+           |
+           |// no val, column of diagnostics must be correct too
+           |decode[Foozz](json)
+           |""".stripMargin
+      )
+      _ <- server.didOpen("main.sc")
+      _ <- server.didSave("main.sc")(identity)
+      _ <- server.executeCommand("ammonite-start")
+
+      _ = server.expectDiagnostics("main.sc", expectedDiagnostics)
+
+    } yield ()
+  }
+
+  test("multi script") {
+    // multiple scripts with mixed import $file-s and $ivy-s
+    for {
+      _ <- server.initialize(
+        s"""
+           |/metals.json
+           |{
+           |  "a": {
+           |    "scalaVersion": "$scalaVersion"
+           |  }
+           |}
+           |/lib1.sc
+           |trait HasFoo {
+           |  def foo: String = "foo"
+           |}
+           |
+           |/lib2.sc
+           |import $$ivy.`io.circe::circe-core:0.12.3`
+           |import $$ivy.`io.circe::circe-generic:0.12.3`
+           |import $$ivy.`io.circe::circe-parser:0.12.3`
+           |import $$file.lib1, lib1.HasFoo
+           |
+           |trait HasReallyFoo extends HasFoo
+           |
+           |/main.sc
+           | // scala $scalaVersion
+           |import $$file.lib2, lib2.HasReallyFoo
+           |import io.circe._, io.circe.generic.auto._, io.circe.parser._, io.circe.syntax._
+           |
+           |sealed trait Foo extends HasReallyFoo
+           |case class Bar(xs: Vector[String]) extends Foo
+           |case class Qux(i: Int, d: Option[Double]) extends Foo
+           |
+           |val foo: Foo = Qux(13, Some(14.0))
+           |
+           |val json = foo.asJson.noSpaces
+           |
+           |val decodedFoo = decode[Foo](json)
+           |""".stripMargin
+      )
+      _ <- server.didOpen("main.sc")
+      _ <- server.didSave("main.sc")(identity)
+      _ <- server.executeCommand("ammonite-start")
+
+      // via Ammonite-generated Semantic DB
+      _ <- server.assertDefinitionAtLocation(
```

Comment:
I would rather do something like
```scala
checkDefinition(
s"""|/main.sc
    |$circeImports
    |val foo: Foo = Qux(13, Some(14.0))
    |
    |val json = foo.asJson.no@@Spaces
    |
    |val decodedFoo = decode[Foo](json)
""",
externalDefinition = Some(".metals/readonly/io/circe/syntax/package.scala")
)
```

and if we have a definition in any of local files I would do:

```scala
checkDefinition(
s"""|/main.sc
    |$circeImports
    |val <<foo>>: Foo = Qux(13, Some(14.0))
    |
    |val json = fo@@o.asJson.noSpaces 
    |
    |val decodedFoo = decode[Foo](json)
"""
)
```
We could most likely reuse some existing method for that. Especially for the second case.

I don't mind having separate case classes for each situation.

Reply:
I'm not sure about that. Sometimes we go-to-definition from `.metals/readonly` files, that wouldn't work with this way of doing it. Also, if we pass files at that point, we would need to import them, so stop / start Ammonite, which I fear would significantly slow things.